src/js/disability-benefits/actions/index.js

```diff
@@ -93,3 +111,136 @@ export function submitRequest(id) {
       .catch(error => dispatch({ type: SET_DECISION_REQUEST_ERROR, error }));
   };
 }
+
+export function getTrackedItem(id) {
+  return (dispatch, getState) => {
+    const item = getState()
+      .claimDetail.detail.attributes.eventsTimeline
+      .filter(event => event.trackedItemId === parseInt(id, 10))[0];
+
+    if (item) {
+      dispatch({
+        type: SET_TRACKED_ITEM,
+        item
+      });
+    }
+  };
+}
+
+export function addFile(files) {
+  return {
+    type: ADD_FILE,
+    files
+  };
+}
+
+export function removeFile(index) {
+  return {
+    type: REMOVE_FILE,
+    index
+  };
+}
+
+export function submitFiles(claimId, trackedItem, files) {
+  let filesComplete = 0;
+  let hasError = false;
+  return (dispatch) => {
+    const uploader = new FineUploaderBasic({
+      request: {
+        endpoint: `${environment.API_URL}/v0/disability_claims/${claimId}/documents`,
+        params: {
+          trackedItem: trackedItem.trackedItemId
+        },
+        inputName: 'file'
+      },
+      cors: {
+        expected: true,
+        sendCredentials: true
+      },
+      multiple: false,
+      callbacks: {
+        onAllComplete: () => {
+          if (!hasError) {
+            dispatch({
+              type: DONE_UPLOADING,
+              itemName: trackedItem.displayName
+            });
+          } else {
+            dispatch({
+              type: SET_UPLOAD_ERROR
+            });
+          }
+        },
+        onComplete: () => {
+          filesComplete++;
+          dispatch({
+            type: SET_PROGRESS,
+            progress: filesComplete / files.length
```

Comment:
The progress info coming back from fine-uploader is a little fast, so I switched to this. Should do some more tweaking.